raiden/raiden_service.py

```diff
@@ -696,271 +1132,323 @@ def on_message(self, message, msghash):  # noqa pylint: disable=unused-argument
-                    # the event handler
-                    self.on_event(originating_contract, event)
-                except:  # pylint: disable=bare-except
-                    log.exception('unexpected exception on log listener')
-
-    def poll_all_event_listeners(self):
-        for event_listener in self.event_listeners:
-            self.poll_event_listener(*event_listener)
+    def dispatch(self, state_manager, state_change):
+        all_events = state_manager.dispatch(state_change)
 
-    def uninstall_listeners(self):
-        chain = self.raiden.chain
+        for event in all_events:
+            self.on_event(event)
 
-        for listener in self.event_listeners:
-            chain.uninstall_filter(listener.filter_.filter_id_raw)
+    def on_event(self, event):
+        if isinstance(event, SendMediatedTransfer):
+            receiver = event.receiver
+            fee = 0
+            graph = self.raiden.channelgraphs[event.token]
+            channel = graph.partneraddress_channel[receiver]
 
-        self.event_listeners = list()
+            mediated_transfer = channel.create_mediatedtransfer(
+                event.initiator,
+                event.target,
+                fee,
+                event.amount,
+                event.identifier,
+                event.expiration,
+                event.hashlock,
+            )
 
-    def on_event(self, emitting_contract_address_bin, event):  # pylint: disable=unused-argument
-        if log.isEnabledFor(logging.DEBUG):
-            log.debug(
-                'event received',
-                type=event['_event_type'],
-                contract=pex(emitting_contract_address_bin),
+            self.raiden.sign(mediated_transfer)
+            channel.register_transfer(mediated_transfer)
+            self.raiden.send_async(receiver, mediated_transfer)
+
+        elif isinstance(event, SendRevealSecret):
+            reveal_message = RevealSecret(event.secret)
+            self.raiden.sign(reveal_message)
+            self.raiden.send_async(event.receiver, reveal_message)
+
+        elif isinstance(event, SendBalanceProof):
+            # TODO: issue #189
+            secret_message = Secret(
+                event.identifier,
+                event.secret,
+                event.token,
             )
+            self.raiden.sign(secret_message)
+            self.raiden.send_async(event.receiver, secret_message)
+
+        elif isinstance(event, SendSecretRequest):
+            secret_request = SecretRequest(
+                event.identifier,
+                event.hashlock,
+                event.amount,
+            )
+            self.raiden.sign(secret_request)
+            self.raiden.send_async(event.receiver, secret_request)
 
-        if event['_event_type'] == 'TokenAdded':
-            self.event_tokenadded(emitting_contract_address_bin, event)
+        elif isinstance(event, SendRefundTransfer):
+            pass
```

Comment:
My bad, didn't finish it, created a issue to finish it: https://github.com/raiden-network/raiden/issues/516